cmd/root.go

```diff
@@ -91,42 +98,57 @@ func WithDialer(d cloudsql.Dialer) Option {
 
 // NewCommand returns a Command object representing an invocation of the proxy.
 func NewCommand(opts ...Option) *Command {
+	cmd := &cobra.Command{}
+
 	c := &Command{
-		conf: &proxy.Config{},
+		Command: cmd,
+		cleanup: func() error { return nil },
+		conf: &proxy.Config{
+			Logger: log.NewStdLogger(os.Stdout, os.Stderr),
+		},
 	}
 	for _, o := range opts {
 		o(c.conf)
 	}
 
-	cmd := &cobra.Command{
-		Use:     "cloud_sql_proxy instance_connection_name...",
-		Version: versionString,
-		Short:   "cloud_sql_proxy provides a secure way to authorize connections to Cloud SQL.",
-		Long: `The Cloud SQL Auth proxy provides IAM-based authorization and encryption when
+	cmd.Version = versionString
+
+	cmd.Use = "cloud_sql_proxy instance_connection_name..."
+	cmd.Short = "cloud_sql_proxy provides a secure way to authorize connections to Cloud SQL."
+	cmd.Long = `The Cloud SQL Auth proxy provides IAM-based authorization and encryption when
 connecting to Cloud SQL instances. It listens on a local port and forwards connections
 to your instance's IP address, providing a secure connection without having to manage
-any client SSL certificates.`,
-		Args: func(cmd *cobra.Command, args []string) error {
-			err := parseConfig(cmd, c.conf, args)
-			if err != nil {
-				return err
-			}
-			// The arguments are parsed. Usage is no longer needed.
-			cmd.SilenceUsage = true
-			return nil
-		},
-		RunE: func(*cobra.Command, []string) error {
-			return runSignalWrapper(c)
-		},
+any client SSL certificates.`
```

Comment:
Seems like we could still declare these inline during initialization on L101

Reply:
Sorry -- trying to make the relationship between cobra.Command and our command more clear here. This probably didn't help. I've cleaned it up.